python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -262,6 +263,28 @@ def _get_ray_resources_from_group_spec(
     if num_gpus is not None:
         resources["GPU"] = num_gpus
 
+    if num_tpus is not None:
+        """Add TPU head resource, similar to the GCP node_provider.
+        Sets the Ray resource TPU-{...}-head to ensure the Ray autoscaler
+        has sufficient resources to make scaling decisions.
+        TPU worker groups treat each TPU podslice as a replica, with `NumOfHosts`
+        specifying the number of workers per slice. Each replica of a TPU worker
+        group has one TPU head.
+
+        For example, a v4-16 worker group with 2 replicas should have the following
```

Comment:
I think the comment here should be updated. 

`worker 0: resources = {"TPU": 4, "TPU-v4-16-head": 2} => worker 0: resources = {"TPU": 4, "TPU-v4-16-head": 1} `

Do we have any issue in the commit history?

Reply:
Fixed in 44aa154. Some of the git history got messed up after a bad rebase but all the changes should be the same.